com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl

```diff
@@ -61,25 +77,54 @@ APVConstants LoadAPVConstants( StructuredBuffer<int> index )
     return apvc;
 }
 
-float3 DecodeSH( float l0, float3 l1 )
+float3 DecodeSH(float l0, float3 l1)
 {
-    return (l1 - 0.5) * 4.0 * l0;
+    // TODO: Enlarging range by 2 because we're using irradiance probes. Need to derive correct factors and remove this.
+    // Compare ProbeGIBaking.cs
+    return (l1 - 0.5f) * 2.0f * l0 * 2.0f;
 }
 
+void DecodeSH_L2(float l0, inout float4 l2_R, inout float4 l2_G, inout float4 l2_B, inout float4 l2_C)
+{
+    l2_R.x = (l2_R.x - 0.5) * 3.75f   * l0;
+    l2_R.y = (l2_R.y - 0.5) * 3.75f   * l0;
+    l2_R.z = (l2_R.z - 0.5) * 3.75f   * l0;
+    l2_R.w = (l2_R.w - 0.5) * 3.75f   * l0;
+                                            
+    l2_G.x = (l2_G.x - 0.5) * 3.75f   * l0;
+    l2_G.y = (l2_G.y - 0.5) * 3.75f   * l0;
+    l2_G.z = (l2_G.z - 0.5) * 0.3125f * l0;
+    l2_G.w = (l2_G.w - 0.5) * 3.75f   * l0;
+                                            
+    l2_B.x = (l2_B.x - 0.5) * 3.75f   * l0;
+    l2_B.y = (l2_B.y - 0.5) * 3.75f   * l0;
+    l2_B.z = (l2_B.z - 0.5) * 0.3125f * l0;
+    l2_B.w = (l2_B.w - 0.5) * 3.75f   * l0;
+                                            
+    l2_C.x = (l2_C.x - 0.5) * 0.9375f * l0;
+    l2_C.y = (l2_C.y - 0.5) * 0.9375f * l0;
+    l2_C.z = (l2_C.z - 0.5) * 0.9375f * l0;
+}
 
 #define APV_USE_BASE_OFFSET
 
 // We split the evaluation in several steps to make variants with different bands easier.
-float3 EvaluateAPVL0(APVResources apvRes, float3 uvw)
+float3 EvaluateAPVL0(APVResources apvRes, float3 uvw, out float L1Rx)
```

Comment:
guess we should also add a 

float3 EvaluateAPVL0(APVResources apvRes, float3 uvw)
{
float unused;
    return EvaluateAPVL0(apvRes, uvw, out unused);
}

in case we only need EvaluateAPVL0 (for volumetric)

Reply:
That is not the public API, the volumetric already have their own simplified "public" API that still works ( https://github.com/Unity-Technologies/Graphics/blob/cc188aeec07e783aeaabab72f3d443cdd7de8a56/com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl#L301 or https://github.com/Unity-Technologies/Graphics/blob/cc188aeec07e783aeaabab72f3d443cdd7de8a56/com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl#L252 if the resources are passed separately)